python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -332,6 +335,27 @@ def process_completed_tasks(topology: Topology) -> None:
             op = active_tasks.pop(ref)
             op.notify_work_completed(ref)
 
+    if active_streaming_gens:
+        ready, _ = ray.wait(
```

Comment:
Probably need to ray.wait() these together with the active_tasks to avoid potential deadlock right?

Reply:
Yup. I plan to convert everything to streaming gen. then there will be only one ray.wait.

Reply:
Found that ActorPoolMapOperator still need to use regular ObjectRefs. Unified them in one single ray.wait.